### PR TITLE
Modify rule S5276: LaYC - narrowing conversions

### PR DESCRIPTION
**Pull request by frederic-tingaud-sonarsource**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S5276/cfamily/rule.adoc [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
 
+Narrowing conversions should be fixed by either using a destination type that can represent all the source values or by using an explicit conversion:
+
+[source,cpp]
+----
+double a = 2.1f; // Compliant: double can represent all floats
+
+long double f();
+double d = 0;
+d += static_cast<double>(f()); // Compliant: clear
```
Comment (amelie-renard-sonarsource) [inline rules/S5276/cfamily/rule.adoc:26]:
```suggestion
d += static_cast<double>(f()); // Compliant: the intent is clear
```

File: rules/S5276/cfamily/rule.adoc [hunk 2]
```diff
@@ -1,19 +1,30 @@
...
-int a = 2.1f; // Noncompliant
+int a = 2.1f; // Noncompliant: loss of floating-point precision
 
 long double f();
 double d = 0;
-d += f(); // Noncompliant
+d += f(); // Noncompliant: smaller range of values
 ----
 
+Narrowing conversions should be fixed by either using a destination type that can represent all the source values or by using an explicit conversion:
```
Comment (amelie-renard-sonarsource) [inline rules/S5276/cfamily/rule.adoc:18]: What about putting the examples and this sentence in the "how to fix it" tab?
I would also add the diff highlighting to the examples

**Review by amelie-renard-sonarsource — CHANGES_REQUESTED** [review]
Nitpicks.
Good idea to add examples in the description, it is far more user-friendly

File: rules/S5276/cfamily/rule.adoc (hunk 2 above)
Comment (frederic-tingaud-sonarsource) [reply rules/S5276/cfamily/rule.adoc:18]: Agreed for the diff, I added it.

Looking at the result of adding a "How to fix it", I have the feeling it is not more readable, so I reverted that.

Comment (amelie-renard-sonarsource) [reply rules/S5276/cfamily/rule.adoc:18]: Ok

**Review by amelie-renard-sonarsource — APPROVED** [review]

**Pull request merged**